groclient/client.py

```diff
@@ -874,13 +877,24 @@ def get_df(self, show_revisions=False, index_by_series=False):
             self.add_points_to_df(
                 None, data_series, self.get_data_points(**data_series)
             )
-        if index_by_series and not self._data_frame.empty:
-            columns = intersect(DATA_SERIES_UNIQUE_TYPES_ID, self._data_frame.columns)
-            if len(columns) > 0:
-                indexed_df = self._data_frame.set_index(columns)
-                indexed_df.index.set_names(DATA_SERIES_UNIQUE_TYPES_ID, inplace=True)
-                return indexed_df.sort_index()
-        return self._data_frame
+
+        df = self._data_frame.copy()
```

Comment:
the data frame can get pretty large so if we make a copy, the client may run out of memory.. How about we just do the name based index in place.

Reply:
fixed